firmitas/base/maintool.py

```diff
@@ -54,49 +54,117 @@ def readcert(certobjc):
+                    "serialno": readdata[7],
+                },
+                "notistat": {
+                    "done": False,
+                    "link": "",
+                    "time": "",
+                }
+            }
+            doneqant += 1
+        except ValueError:
+            logrdata.logrobjc.error(
+                f"[{file.stem}] The specified X.509-standard TLS certificate could not be read"
+            )
+            failqant += 1
+
+    logrdata.logrobjc.info(
+        f"Of {totlqant} TLS certificates, {doneqant} TLS certificate(s) were read successfully "
+        + f"while {failqant} TLS certificate(s) could not be read"
+    )
+
+    with open(standard.hostloca, "w") as yamlfile:
+        yaml.safe_dump(standard.certdict, yamlfile)
+
+
 def probedir():
     logrdata.logrobjc.info("Probing into the configured directory")
     doneqant, failqant, totlqant = 0, 0, 0
+
+    logrdata.logrobjc.info("Validating X.509-standard TLS certificates")
     standard.certdict = yaml.safe_load(Path(standard.hostloca).read_text())
-    logrdata.logrobjc.info(
-        f"Validating {len(standard.certdict)} X.509-standard TLS certificates"
-    )
+
     for nameindx in standard.certdict:
+        certpath = Path(standard.certdict[nameindx]["path"])
         totlqant += 1
-        certpath = Path(standard.certloca, standard.certdict[nameindx]["path"])
-        if os.path.exists(certpath):
-            try:
-                certobjc = x509.load_pem_x509_certificate(certpath.read_bytes(), default_backend())
-                (
-                    standard.certdict[nameindx]["certstat"]["strtdate"],
-                    standard.certdict[nameindx]["certstat"]["stopdate"],
-                    standard.certdict[nameindx]["certstat"]["cstarted"],
-                    standard.certdict[nameindx]["certstat"]["cstopped"],
-                    standard.certdict[nameindx]["certstat"]["daystobt"],
-                    standard.certdict[nameindx]["certstat"]["daystodd"],
-                    standard.certdict[nameindx]["certstat"]["issuauth"],
-                    standard.certdict[nameindx]["certstat"]["serialno"],
-                ) = readcert(certobjc)
-                doneqant += 1
-                logrdata.logrobjc.info(
-                    f"[{nameindx}] The specified X.509-standard TLS certificate was read "
-                    + "successfully"
-                )
-            except ValueError:
-                failqant += 1
-                logrdata.logrobjc.error(
-                    f"[{nameindx}] The specified X.509-standard TLS certificate could not be read"
-                )
-        else:
-            failqant += 1
+
+        if not os.path.exists(certpath):
             logrdata.logrobjc.warning(
                 f"[{nameindx}] The specified X.509-standard TLS certificate could not "
                 + "be located"
             )
+            failqant += 1
+            continue
+
+        try:
+            print(certpath)
```

Comment:
I guess this is a stray debugging line, should it go in?

Reply:
Exactly - I will remove it.